plugins/modules/vmware_guest_instant_clone.py

```diff
@@ -0,0 +1,397 @@
+        # datacentre check
+        if self.datacenter is None:
+            self.module.fail_json(msg="Datacenter not found.")
+
+        datastore_name = self.params['datastore']
+        datastore_cluster = find_obj(self.destination_content, [vim.StoragePod], datastore_name)
+
+        if datastore_cluster:
+            # If user specified datastore cluster so get recommended datastore
+            datastore_name = self.get_recommended_datastore(datastore_cluster_obj=datastore_cluster)
+            # Check if get_recommended_datastore or user specified datastore exists or not
+        # datastore check
+        self.datastore = find_datastore_by_name(content=self.destination_content, datastore_name=datastore_name)
+
+        if self.datastore is None:
+            self.module.fail_json(msg="Datastore not found.")
+
+        if self.params['folder']:
+            self.folder = find_folder_by_name(content=self.destination_content, folder_name=self.params['folder'])
+            if self.folder is None:
+                self.module.fail_json(msg="Folder not found.")
+        else:
+            self.folder = self.datacenter.vmFolder
+
+        self.host = find_hostsystem_by_name(content=self.destination_content, hostname=self.params['host'])
+        if self.host is None:
+            self.module.fail_json(msg="Host not found.")
+
+        if self.params['resource_pool']:
+            self.resource_pool = find_resource_pool_by_name(
+                content=self.destination_content,
+                resource_pool_name=self.params['resource_pool'])
+            if self.resource_pool is None:
+                self.module.fail_json(msg="Resource Pool not found.")
+        else:
+            self.resource_pool = self.host.parent.resourcePool
+
+    def populate_specs(self):
+
+        # populate relocate spec
+        self.relocate_spec.datastore = self.datastore
+        self.relocate_spec.pool = self.resource_pool
+        self.relocate_spec.folder = self.folder
+        # populate Instant clone spec
+        self.instant_clone_spec.name = self.vm_name
+        self.instant_clone_spec.location = self.relocate_spec
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        name=dict(type='str', required=True, aliases=['vm_name']),
+        uuid=dict(type='str'),
+        moid=dict(type='str'),
+        datacenter=dict(type='str', required=True),
+        datastore=dict(type='str', required=True),
+        use_instance_uuid=dict(type='bool', default=False),
+        hostname=dict(type='str', aliases=['admin', 'user']),
+        host=dict(type='str', required=True),
+        folder=dict(type='str', required=False),
+        resource_pool=dict(type='str', required=False),
+        parent_vm=dict(type='str'),
+        port=dict(type='int', default=443),
+        username=dict(type='str'),
+        password=dict(type='str', no_log=True, aliases=['pass', 'pwd']),
+        validate_certs=dict(type='bool', default=False)
```

Comment:
```suggestion
```

There are defined(include the `hostname`) in vmware_argument_spec (vmware.py), so I think these are unnecessary.
Can you please let me know if there is a reason to overwrite these?

Reply:
These sure look unnecessary, thank you and I will change them.